app/pages/challenges/components/challengeCard.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import { Card, CardHeader, CardTitle, CardDescription, CardContent, CardFooter } from "@/components/ui/card";
+import { Progress } from "@/components/ui/progress";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+
+interface Challenge {
+    id: string;
+    title: string;
+    description: string;
+    totalProgress: number;
+    goal: number;
+    unit: string;
+    leaderboard: { name: string; score: number }[];
+}
+
+interface ChallengeCardProps {
```

Comment:
very nice but why do you need to have this type if it is already created in app/api/models